src/components/tree-views-container.js

```diff
@@ -47,6 +86,196 @@ const TreeViewsContainer = () => {
 
     const { enqueueSnackbar } = useSnackbar();
 
+    /* Dialogs states */
+    const [anchorEl, setAnchorEl] = React.useState(null);
+    const [openAddNewStudyDialog, setOpenAddNewStudyDialog] = React.useState(
+        false
+    );
+    const [
+        openCreateNewDirectoryDialog,
+        setOpenCreateNewDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openDeleteDirectoryDialog,
+        setOpenDeleteDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openCreateRootDirectoryDialog,
+        setOpenCreateRootDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openRenameDirectoryDialog,
+        setOpenRenameDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openAccessRightsDirectoryDialog,
+        setOpenAccessRightsDirectoryDialog,
+    ] = React.useState(false);
+
+    const [accessRightsError, setAccessRightsError] = React.useState('');
+    const [deleteError, setDeleteError] = React.useState('');
+    const [renameError, setRenameError] = React.useState('');
+
+    /* Menu states */
+    const [mousePosition, setMousePosition] = React.useState(
+        initialMousePosition
+    );
+
+    const [
+        showCreateStudyMenuItem,
+        setShowCreateStudyMenuItem,
+    ] = React.useState(true);
+    const [
+        showCreateNewDirectoryMenuItem,
+        setShowCreateNewDirectoryMenuItem,
+    ] = React.useState(true);
+
+    /* User interactions */
+    const onContextMenu = useCallback(
+        (e, nodeId) => {
+            setMousePosition({
+                mouseX: e.clientX + 16,
```

Comment:
maybe create constants to hold 16 and 4 values

Reply:
ok